src/mp4atom.cpp

```diff
@@ -235,11 +235,17 @@ void MP4Atom::Read()
                      m_File.GetFilename().c_str(), m_type, m_size);
     }
 
-    ReadProperties();
-
-    // read child atoms, if we expect there to be some
-    if (m_pChildAtomInfos.Size() > 0) {
-        ReadChildAtoms();
+    // skip parsing of certain atoms
+    ShouldParseAtomCallback cb = m_File.GetShouldParseAtomCallback();
+    if ( cb == nullptr || ( cb != nullptr && cb( ATOMID(m_type) ) ) )
+    {
```

Comment:
nitpick suggestion:  (add explanatory variable, and don't check `cb!=nullptr` because it's redundant)

ShouldParseAtomCallback cb = m_File.GetShouldParseAtomCallback();
bool shouldParseAtom = cb == nullptr || cb( ATOMID(m_type) );
if ( shouldParseAtom )
...